### PR TITLE
Fixes #1140 for reports

### LINKED ISSUE #1140: Enhancement: limit report to updates actionable in the context of pom.xml
I found nothing similar in closed issues, if you consider this a duplicate to a previous issue, feel free to close it.

**Motivation:**
I'm trying to limit the reports down to a point that all listed dependencies, plugins, parent and properties version are specified in the analyzed pom.xml. Basically all inherited (upstream) issues should be ignored. I want to make it as simple and fast as possible to fix such issues without the need to reason about if a reported issue is inherited from upstream.

**Example**: two projects, from different reactors

* parent/pom.xml uses spring-boot-starter-parent:3.3.1, but 3.3.3 is latest version
* service/pom.xml uses parent/pom.xml at latest version

report for
* parent/pom.xml 
  * MUST list outdated parent
  * MUST NOT list e.g. outdated plugins, dependencies that are inherited from spring-boot-starter-parent
* service/pom.xml
  * MUST NOT list e.g. outdated parent, plugins etc. inherited from parent/pom.xml

**Possible Solutions**:

* preferred, additional flag e.g. _ignoreInherited_ to control this behavior
* alternative, mark "inherited" issues in reports (and possibly xml) as such

### PR DESCRIPTION
**Pull request by andrzejj0**
I managed to reproduce #1140 for reports, so here is a fix.

Added an integration test + showVersionless for dependency-updates-report.

Added some small refactoring.

### CODE REVIEW

File: versions-maven-plugin/src/main/java/org/codehaus/mojo/versions/AbstractDependencyUpdatesReport.java [hunk 1]
```diff
@@ -80,6 +81,17 @@ public abstract class AbstractDependencyUpdatesReport extends AbstractVersionsRe
     @Parameter(property = "processDependencyManagementTransitive", defaultValue = "true")
     protected boolean processDependencyManagementTransitive;
 
+    /**
+     * <p>Include dependencies with version set in a parent or in a BOM.</p>
+     * <p>This is similar to {@code processDependencyManagementTransitive}, but will
+     * report updates on dependencies.</p>
+     * <p>Default is {@code false}.</p>
```
Comment (slawekjaranowski) [inline versions-maven-plugin/src/main/java/org/codehaus/mojo/versions/AbstractDependencyUpdatesReport.java:88]: When we use `defaultValue` in `@Parameter` is will be added to docs.
By the way has different value .... doc and parameter

Comment (andrzejj0) [reply versions-maven-plugin/src/main/java/org/codehaus/mojo/versions/AbstractDependencyUpdatesReport.java:88]: I get your point, but I did not change `processDependencyManagementTransitive` in this PR.

On the other hand, this (initializing the properties) immensely helps with unit testing.

File: versions-maven-plugin/src/main/java/org/codehaus/mojo/versions/AbstractDependencyUpdatesReport.java [hunk 2]
```diff
@@ -80,6 +81,17 @@ public abstract class AbstractDependencyUpdatesReport extends AbstractVersionsRe
     @Parameter(property = "processDependencyManagementTransitive", defaultValue = "true")
     protected boolean processDependencyManagementTransitive;
 
+    /**
+     * <p>Include dependencies with version set in a parent or in a BOM.</p>
+     * <p>This is similar to {@code processDependencyManagementTransitive}, but will
+     * report updates on dependencies.</p>
+     * <p>Default is {@code true}.</p>
```
Comment (slawekjaranowski) [inline versions-maven-plugin/src/main/java/org/codehaus/mojo/versions/AbstractDependencyUpdatesReport.java:88]: it is not needed will be generated .... check site output 😄

Comment (andrzejj0) [reply versions-maven-plugin/src/main/java/org/codehaus/mojo/versions/AbstractDependencyUpdatesReport.java]: Removed <p>Default is {@code true}.</p>

File: versions-maven-plugin/src/main/java/org/codehaus/mojo/versions/AbstractDependencyUpdatesReport.java [hunk 3]
```diff
@@ -80,6 +81,16 @@ public abstract class AbstractDependencyUpdatesReport extends AbstractVersionsRe
     @Parameter(property = "processDependencyManagementTransitive", defaultValue = "true")
     protected boolean processDependencyManagementTransitive;
 
+    /**
+     * <p>Include dependencies with version set in a parent or in a BOM.</p>
+     * <p>This is similar to {@code processDependencyManagementTransitive}, but will
+     * report updates on dependencies.</p>
+     *
+     * @since 2.18.1
```
Comment (slawekjaranowski) [inline versions-maven-plugin/src/main/java/org/codehaus/mojo/versions/AbstractDependencyUpdatesReport.java:89]: Maybe should be a 2.19.0 as we add new feature ... what do you think?

Comment (slawekjaranowski) [reply versions-maven-plugin/src/main/java/org/codehaus/mojo/versions/AbstractDependencyUpdatesReport.java:89]: or it is for fixing bug only ...

Comment (andrzejj0) [reply versions-maven-plugin/src/main/java/org/codehaus/mojo/versions/AbstractDependencyUpdatesReport.java]: Ok. Changed that in DisplayDependencyUpdatesMojo as well.

Comment (andrzejj0) [reply versions-maven-plugin/src/main/java/org/codehaus/mojo/versions/AbstractDependencyUpdatesReport.java]: Well, a bug fix and a feature toggle. I don't know :-)

**Pull request merged**